ci/infra/testrunner/README.md

```diff
@@ -30,6 +31,56 @@
 
 Testrunner is a CLI tool for setting up an environment for running e2e tests, making transparent the mechanism used for providing the test infrastructure. It can be used as a stand-alone tool, running either locally or as part of a CI pipeline. It provides commands for provisioning the infrastructure, deploying a k8s cluster using `skuba`,  and running tests. It also provides [a library](tests/README.md) for developing `pytest`-based tests.
 
+
+## Design
+
+The `testrunner` is composed by multiple components as described in the figure below:
+* The `testrunner.py` frontend cli application, which receives CLI options, loads a configuration and executes the selected command. The configuration is loaded from a yaml file (by default, [vars.yaml](vars.yaml)) and initialized by the `BaseConfig` class (located in [utils/constanst.py](utils/constanst.py) which mixes it with the environment variables and sets defaults values for missing configuration parameters.   
```

Comment:
Do you think it provides value to go that detailed about the `BaseConfig` class?

Reply:
In practice I've found most people is not aware of this class and how it works, mostly related to how it sets defaults and merge configuration. I actually would like to rename the module to `config.py` to make more evident its intention.

Reply:
I'm removing this detail, but I think it should be documented somewhere, maybe a section on how to add new configuration parameters.